### PR TITLE
Moved tests into child folders Unit and Integration

### PR DESCRIPTION
**Pull request by Dan1el42**
1. Moved Unit tests into the child folder Unit
2. Updated moved *.Tests.ps1 to find and load the resource modules again.
3. Moved MSFT_xWebBindingInformation.Tests.ps1 into the Integration child folder because it creates a temporary configuration and uses the LCM to config a "foobar" web site. I would classify this test behavior as integration not unit.
4. Simplified how the tests reference the resource module names and paths.

### CODE REVIEW

**Comment by tysonjhayes** [comment]
So this actually fails in my lab, running WMF 4.0 (I believe AppVeyor is running WMF 5.0 but someone would need to correct me if I'm wrong there).

Currently it fails with this error:

````
Describing Schema Validation MSFT_xWebsite
 [-] should pass Test-xDscResource 82ms
   Cannot bind argument to parameter 'Name' because it is an empty string.
   at line: 20 in D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebsite.Tests.ps1
Join-Path : Cannot bind argument to parameter 'Path' because it is null.
At D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebsite.Tests.ps1:26 char:13
+             Join-Path -ChildPath "$($DSCResource.Name).schema.mof" -Resolve
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Join-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.JoinPathCommand

 [-] should pass Test-xDscSchema 16ms
   Cannot bind argument to parameter 'Path' because it is an empty string.
   at line: 27 in D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebsite.Tests.ps1
[-] Error occurred in test script 'D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebsite.Tests.ps1' 30ms
  Cannot bind argument to parameter 'Name' because it is null.
  at <ScriptBlock>, D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebsite.Tests.ps1: line 34
Describing Schema Validation MSFT_xWebVirtualDirectory
 [-] should pass Test-xDscResource 77ms
   Cannot bind argument to parameter 'Name' because it is an empty string.
   at line: 19 in D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebVirtualDirectory.tests.ps1
Join-Path : Cannot bind argument to parameter 'Path' because it is null.
At D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebVirtualDirectory.tests.ps1:25 char:13
+             Join-Path -ChildPath "$($DSCResource.Name).schema.mof" -Resolve
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Join-Path], ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Microsoft.PowerShell.Commands.JoinPathCommand

 [-] should pass Test-xDscSchema 16ms
   Cannot bind argument to parameter 'Path' because it is an empty string.
   at line: 26 in D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebVirtualDirectory.tests.ps1
[-] Error occurred in test script 'D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebVirtualDirectory.tests.ps1' 30ms
  Cannot bind argument to parameter 'Name' because it is null.
  at <ScriptBlock>, D:\GitHub\tysonjhayes\Deploy\xWebAdministration\tests\Unit\MSFT_xWebVirtualDirectory.tests.ps1: line 33
Tests completed in 253ms
Passed: 0 Failed: 6 Skipped: 0 Pending: 0
````

Breaking into the debugger I'm finding that $DSCResource is null, specifically $PSScriptRoot.  I believe I've run into this before which is why I coded in the $here variable (which in hindsight was nomenclature picked up from Pester but is a terribly named variable name).

Also Piping is inherantly slower to process and where you are using isn't gaining us anything. I would suggest not using the pipe and just calling the variables on the correct parameter.

File: Tests/Integration/MSFT_xWebBindingInformation.Tests.ps1 [hunk 1]
```diff
@@ -1,51 +1,54 @@
...
-    $storeNames = (Get-CimClass -Namespace "root/microsoft/Windows/DesiredStateConfiguration" -ClassName "MSFT_xWebBindingInformation").CimClassProperties['CertificateStoreName'].Qualifiers['Values'].Value
-    foreach ($storeName in $storeNames){
-        It "Uses valid credential store: $storeName" {
-            (Join-Path -Path Cert:\LocalMachine -ChildPath $storeName) | Should Exist
-        }
-    }
-}
+﻿$DSCResourceName = 'MSFT_xWebsite'
+$DSCResource = Get-Module -ListAvailable -Name (
+    $PSScriptRoot | Join-Path -ChildPath "..\..\DSCResources\$DSCResourceName\$DSCResourceName.psm1"
```
Comment (tysonjhayes) [inline Tests/Integration/MSFT_xWebBindingInformation.Tests.ps1]: This throws an error (see other comment). I would suggest changing to this:

$DSCResource = Get-Module -ListAvailable -Name (
    Join-Path -Path $(Split-Path -Parent $MyInvocation.MyCommand.Path) -ChildPath "..\..\DSCResources\$DSCResourceName\$DSCResourceName.psm1"
)

File: Tests/Integration/MSFT_xWebBindingInformation.Tests.ps1 [hunk 2]
```diff
@@ -1,51 +1,54 @@
...
+)
+
+# should check for the server OS
+if($env:APPVEYOR_BUILD_VERSION)
+{
+  Add-WindowsFeature Web-Server -Verbose
+}
+
+$DSCResource | Import-Module -Force
```
Comment (tysonjhayes) [inline Tests/Integration/MSFT_xWebBindingInformation.Tests.ps1]: The pipe here isn't giving us anything. Could we please change this to

````
Import-Module -Name $DSCResource -Force
````

File: Tests/Integration/MSFT_xWebBindingInformation.Tests.ps1 (hunk 1 above)
Comment (tysonjhayes) [inline Tests/Integration/MSFT_xWebBindingInformation.Tests.ps1]: This basically applies every where we are calling $DSCResource but I didn't want to spam the pull request pointing it out. :)

**Comment by tysonjhayes** [comment]
Thanks for the PR! I look forward to working with you on getting these changes in.

**Comment by Dan1el42** [comment]
Thanks for the feedback. I will fix it and update the PR.

**Comment by Dan1el42** [comment]
I think I have managed fixed the errors I introduced on WMF v4 and incorporate your suggestion to not use the pipeline. Please be so kind to test again in your lab.

**Comment by TravisEz13** [comment]
Changes look good, waiting on tyson to confirm it works on WMF 4 to merge

**Comment by tysonjhayes** [comment]
Works great. Thanks @Dan1el42!

**Pull request merged**